src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1159,6 +1159,11 @@ object Types {
     private[this] var lastSymbol: Symbol = _
     private[this] var checkedPeriod = Nowhere
 
+    /** Provided for debugging, can be used to intercept assignments of uncommon denotations
+     */  
+    private def checkDenot(d: Denotation)(implicit ctx: Context) = {
```

Comment:
Well, I to find out about the problems with unpickling I needed checkDenot, because I needed to intercept every setting of lastDenot and I needed a context. But it's no big deal; we can delete the method if you feel it is redundant.